### PR TITLE
chore: editable email chip merchant requirement

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by Riddhiagrawal001** [comment]
The main issue persists even after making the chip editable: the "Add User" button remains disabled.
image

File: src/components/form/MultipleTextInput.res [hunk 1]
```diff
@@ -50,8 +50,23 @@ let make = (
   }
 
   let (text, setText) = React.useState(_ => "")
+  let (makeChip, setMakeChip) = React.useState(_ => false)
+  let (shiftPressed, setShiftPressed) = React.useState(_ => false)
+  let (invalidEmail, setInvalidEmail) = React.useState(_ => false)
+
+  React.useEffect1(() => {
```
Comment (sagarnaikjuspay) [inline src/components/form/MultipleTextInput.res:57]: React.useEffect can be used

File: src/components/form/MultipleTextInput.res [hunk 2]
```diff
@@ -66,6 +81,13 @@ let make = (
   })
   let handleKeyDown = e => {
     open ReactEvent.Keyboard
+
```
Comment (sagarnaikjuspay) [inline src/components/form/MultipleTextInput.res:84]: key value can be extracted once and used in if condition

File: src/components/form/MultipleTextInput.res [hunk 3]
```diff
@@ -103,10 +127,25 @@ let make = (
           })
 
           setTags(currentTags->Array.concat(newArr))
+          setMakeChip(_ => false)
+          setText(_ => "")
         } else if !(currentTags->Array.includes(text->String.trim)) {
-          setTags(currentTags->Array.concat([text->String.trim]))
+          let a = text->String.trim->LogicUtils.isValidEmail
```
Comment (sagarnaikjuspay) [inline src/components/form/MultipleTextInput.res:133]: can u use proper variable name here

File: src/components/form/MultipleTextInput.res [hunk 4]
```diff
@@ -66,6 +81,13 @@ let make = (
...
   let handleKeyDown = e => {
     open ReactEvent.Keyboard
+
+    if e->key === "Shift" {
+      setShiftPressed(_ => true)
+    } else if e->key != "2" {
+      setShiftPressed(_ => false)
+    }
+
     let isEmpty = text->LogicUtils.isEmptyString
```
Comment (sagarnaikjuspay) [inline src/components/form/MultipleTextInput.res:91]: opne LogicUtils once at top

File: src/components/form/MultipleTextInput.res [hunk 5]
```diff
@@ -136,25 +175,43 @@ let make = (
...
-    <TextInput
-      input=input1
-      focusOnKeyPress={keyDownCondition}
-      placeholder
-      ?autoComplete
-      onKeyUp=handleKeyDown
-      isDisabled=disabled
-      customStyle="dark:bg-jp-gray-970 border-none"
-    />
+                  border rounded border-opacity-75  dark:border-jp-gray-960 ${invalidEmail
```
Comment (sagarnaikjuspay) [inline src/components/form/MultipleTextInput.res:178]: move this condition to extarnal variable

File: src/utils/LogicUtils.res [hunk 6]
```diff
@@ -676,3 +676,9 @@ let getValFromNullableValue = (val, default) => {
 }
 
 let dateFormat = (timestamp, format) => (timestamp->DayJs.getDayJsForString).format(format)
+
+let isValidEmail = value =>
+  !Js.Re.test_(
```
Comment (sagarnaikjuspay) [inline src/utils/LogicUtils.res:681]: check this Regex in safari once

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request closed without merging**